src/identity/interaction/routing/BasicInteractionRoute.ts

```diff
@@ -1,101 +1,43 @@
-  InteractionHandler,
-  Interaction,
-} from '../InteractionHandler';
-import type { InteractionRoute, TemplatedInteractionResult } from './InteractionRoute';
+import type { Representation } from '../../../http/representation/Representation';
+import { NotFoundHttpError } from '../../../util/errors/NotFoundHttpError';
+import { UnsupportedAsyncHandler } from '../../../util/handlers/UnsupportedAsyncHandler';
+import { InteractionHandler } from '../InteractionHandler';
+import type { InteractionHandlerInput } from '../InteractionHandler';
+import type { InteractionRoute } from './InteractionRoute';
 
 /**
- * Default implementation of the InteractionRoute.
- * See function comments for specifics.
+ * Default implementation of an InteractionHandler with an InteractionRoute.
+ *
+ * Rejects operations that target a different path,
+ * otherwise the input parameters get passed to the source handler.
+ *
+ * In case no source handler is provided it defaults to an {@link UnsupportedAsyncHandler}.
+ * This can be useful if you want an object with just the route.
  */
-export class BasicInteractionRoute implements InteractionRoute {
-  public readonly route: RegExp;
-  public readonly handler: InteractionHandler;
-  public readonly viewTemplates: Record<string, string>;
-  public readonly prompt?: string;
-  public readonly responseTemplates: Record<string, string>;
-  public readonly controls: Record<string, string>;
+export class BasicInteractionRoute extends InteractionHandler implements InteractionRoute {
+  private readonly path: string;
+  private readonly source: InteractionHandler;
 
-  /**
-   * @param route - Regex to match this route.
-   * @param viewTemplates - Templates to render on GET requests.
-   *                        Keys are content-types, values paths to a template.
-   * @param handler - Handler to call on POST requests.
-   * @param prompt - In case of requests to the IDP entry point, the session prompt will be compared to this.
-   * @param responseTemplates - Templates to render as a response to POST requests when required.
-   *                            Keys are content-types, values paths to a template.
-   * @param controls - Controls to add to the response JSON.
-   *                   The keys will be copied and the values will be converted to full URLs.
-   */
-  public constructor(route: string,
-    viewTemplates: Record<string, string>,
-    handler: InteractionHandler,
-    prompt?: string,
-    responseTemplates: Record<string, string> = {},
-    controls: Record<string, string> = {}) {
-    this.route = new RegExp(route, 'u');
-    this.viewTemplates = viewTemplates;
-    this.handler = handler;
-    this.prompt = prompt;
-    this.responseTemplates = responseTemplates;
-    this.controls = controls;
+  public constructor(path: string, source?: InteractionHandler) {
+    super();
+    this.path = path;
+    this.source = source ?? new UnsupportedAsyncHandler('This route has no associated handler.');
```

Comment:
When does that error case occur?

Reply:
This is currently used by the HTML response pages, since we need a route to those but they're not tied to an actual handler. So this behaviour could disappear if we get rid of the response pages and do everything in the JS of the input page. Although we might still want it in the future if we have a register page but no longer a simple registration API.